Compute DiskRP release

grizzlytheodore opened the pull request:
This update is generated from 3 changes in Compute: 
1. Compute's migration to TypeSpec  https://github.com/Azure/azure-rest-api-specs/pull/32748
2. New DiskRP version release https://github.com/Azure/azure-rest-api-specs/pull/35205
3. Breaking change - updating GetVMImages with Properties operation to return array instead of pageable to correctly reflect what the backend is returning https://github.com/Azure/azure-rest-api-specs/pull/35509

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

xinzewang233 committed:
update autorest commit id

Commit:
generated with new swagger references

Commit:
clean up autorest directives

grizzlytheodore commented on sdk/compute/Azure.ResourceManager.Compute/src/Generated/VirtualMachineScaleSetVmData.cs:
```diff
@@ -66,20 +66,20 @@ public VirtualMachineScaleSetVmData(AzureLocation location) : base(location)
         /// <param name="etag"> Etag is property returned in Update/Get response of the VMSS VM, so that customer can supply it in the header to ensure optimistic updates. </param>
         /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
-        internal VirtualMachineScaleSetVmData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, IDictionary<string, string> tags, AzureLocation location, string instanceId, ComputeSku sku, VirtualMachineScaleSetVmProperties properties, ComputePlan plan, IReadOnlyList<VirtualMachineExtensionData> resources, IReadOnlyList<string> zones, ManagedServiceIdentity ide ...
+        internal VirtualMachineScaleSetVmData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, IDictionary<string, string> tags, AzureLocation location, VirtualMachineScaleSetVmProperties properties, string instanceId, ComputeSku sku, ComputePlan plan, IReadOnlyList<VirtualMachineExtensionData> resources, IReadOnlyList<string> zones, ManagedServiceIdentity ide ...
```
@ArcturusZhang Can you help me understand why this ordering is changing? 

For track 2, even if the orders in swagger change, it should resolve itself right? 

In this PR: https://github.com/Azure/azure-sdk-for-net/pull/48196/files#diff-0b9e8253924ab02a897273a24840c766130e6dc542fdf07066f71c0a5fbd95be
it is referencing the same object as this one, but that PR generated the parameters order that was same as before an in this PR, the param signature changed. 

trying to understand what would cause that

ArcturusZhang replied:
First of all, in track 2 generator, we did some tricks but in general, in the same model if you change the order of properties, the generated code would change.
The change of the order of properties would not be a breaking change with itself, but since we use that order to generate the methods in model factory, which makes the order of properties into the order of parameters. The change of order of parameters is breaking change.
Therefore in this case, the parameter order here changed because the property order changes.
Your new swagger is generated from typespec, right?
In typespec, the model represents the properties in a different way if you recall - and the swagger emitter would honor that recursively.
The model of a resource comes in in this way:
```
model VirtualMachineScaleSetVm is TrackedResource<VirtualMachineScaleSetVmProperties>
{
	...SkuProperty;
}
```
you could see the property `properties` comes above the property `sku` in typespec. This is the root cause of the property order change.

This is not a new issue - we knew it along the way of migration from swagger to typespec.
The current generator autorest.csharp has been feature frozen for a while therefore this "old" generator would not be able to automatically handle it, but we are making a new mgmt generator based on new typespec stuff, in which this breaking change if happened again in the future, will be automatically handled (by reading the parameter order from the contract and keep it unchanged)

But for now, the only way out for us, because the new generator is not ready enough to use right now, is to write some customization code to mitigate this change.

Commit:
fix changed signature with customization

Commit:
export api, update snipper

Commit:
resolve breaking change and suppress intentional breaking change

Commit:
update test references and version ref

Commit:
add changelog

Commit:
add assets.json

grizzlytheodore commented:
@ArcturusZhang I suppressed them. Please review

ArcturusZhang commented:
> @ArcturusZhang
> 
> I am getting these two build errors: ![image](https://private-user-images.githubusercontent.com/10857691/463804900-c275f6be-c612-4866-a447-617f4f81ba76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIwMzkzMzUsIm5iZiI6MTc1MjAzOTAzNSwicGF0aCI6Ii8xMDg1NzY5MS80NjM4MDQ5MDAtYzI3NWY2YmUtYzYxMi00ODY2LWE0NDctNjE3ZjRmODFiYTc2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA5VDA1MzAzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2ZjU5NTY1OTZjNGE0NDM4MDY0OGU1MTk1YThiYzNkYjkxOTA0NTE5NjUzZjZmY2Q0OGE1ZTg2ZDU4Mzk5YzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0_5xNTLnVW7L_F9AwwPrQxMqeJRfq9LirTHaMOJTfog)
> 
> which comes from breaking change we made in typespec/swagger: [Azure/azure-rest-api-specs#35509](https://github.com/Azure/azure-rest-api-specs/pull/35509) It was to correctly reflect what is being returned from the backend. Before it was defined as pageable response, but the service side actually returns an array.
> 
> Can I suppress this? How can I suppress this?

We have this breaking change because in your linked swagger PR, we removed the `Expand` enum and changed its type to a string.
The response change actually does not trigger a breaking change because the generator will treat APIs that returns an array as a pageable operation.

Therefore we just need to resolve the issue of parameter type change.

grizzlytheodore commented on sdk/compute/Azure.ResourceManager.Compute/src/Customize/ArmComputeModelFactory.cs:
```diff
@@ -225,6 +225,31 @@ public static VirtualMachineScaleSetVmData VirtualMachineScaleSetVmData(Resource
         }
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static VirtualMachineScaleSetVmData VirtualMachineScaleSetVmData(ResourceIdentifier id = null, string name = null, ResourceType resourceType = default, SystemData systemData = null, IDictionary<string, string> tags = null, AzureLocation location = default, string instanceId = null, ComputeSku sku = null, VirtualMachineScaleSetVmProperties properties = null, ComputePlan plan = null, ...
```
made the properties until VirtualMachineScaleSetVmProperties properties = null 
required. which is where the switch up happened

Commit:
made custom function parameters required

Commit:
make more required parameters - missed params

Commit:
remove suppresion and add customization for brekaing change mitigation

grizzlytheodore commented:
> > @ArcturusZhang
> > I am getting these two build errors: ![image](https://private-user-images.githubusercontent.com/10857691/463804900-c275f6be-c612-4866-a447-617f4f81ba76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIwMzkzMzUsIm5iZiI6MTc1MjAzOTAzNSwicGF0aCI6Ii8xMDg1NzY5MS80NjM4MDQ5MDAtYzI3NWY2YmUtYzYxMi00ODY2LWE0NDctNjE3ZjRmODFiYTc2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA5VDA1MzAzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2ZjU5NTY1OTZjNGE0NDM4MDY0OGU1MTk1YThiYzNkYjkxOTA0NTE5NjUzZjZmY2Q0OGE1ZTg2ZDU4Mzk5YzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0_5xNTLnVW7L_F9AwwPrQxMqeJRfq9LirTHaMOJTfog)
> > which comes from breaking change we made in typespec/swagger: [Azure/azure-rest-api-specs#35509](https://github.com/Azure/azure-rest-api-specs/pull/35509) It was to correctly reflect what is being returned from the backend. Before it was defined as pageable response, but the service side actually returns an array.
> > Can I suppress this? How can I suppress this?
> 
> We have this breaking change because in your linked swagger PR, we removed the `Expand` enum and changed its type to a string. The response change actually does not trigger a breaking change because the generator will treat APIs that returns an array as a pageable operation.
> 
> Therefore we just need to resolve the issue of parameter type change.

I removed the ApiCompatBaseline.txt suppresions
and added 2 customizations to resolve 2/3 MembersMustExist error: 


I still have this issue and I am not sure how to resolve it: 
![image](https://github.com/user-attachments/assets/eb8a1dcb-4a7a-472a-ac3e-461aadb6285d)

ArcturusZhang committed:
fix issues

grizzlytheodore commented on sdk/compute/Azure.ResourceManager.Compute/src/autorest.md:
```diff
@@ -390,6 +390,11 @@ directive:
+  - from: ComputeRP.json
+    where: $["x-ms-paths"]["/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions?$expand=Properties"].get
+    transform: >
+      $.parameters[6]["x-ms-client-name"] = "expandOption";
```
I see the approach. very smart!

Commit:
Merge branch 'feature/cplat-2025-01-02' of https://github.com/Azure/azure-sdk-for-net into feature/cplat-2025-01-02

ArcturusZhang commented on sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md:
```diff
@@ -1,16 +1,22 @@
-### Other Changes
+- Removed `GetVirtualMachineImagesWithPropertiesExpand` class as it has no utility. 
 
-## 1.10.0 (2026-06-30)
```
LOL I did this, sorry about that
